### PR TITLE
add postgres serviceUpdates configuration

### PR DESCRIPTION
**Pull request by laurafitzgerald**
# Issue link
https://issues.redhat.com/browse/MGDAPI-3095

# What
Specify serviceUpdate configuration for Postgres
Related pr -> https://github.com/integr8ly/cloud-resource-operator/pull/235

# Verification steps
Install workloadwebapp in the cluster to verify downtime during the upgrade

Checkout the master branch - update products/installation.yaml cro reference to point to -> quay.io/laurafitzgerald/cro-index:v0.30.0
Run `INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make code/run`
Allow the installation to go completed

Log into the AWS account, verify that there are 3 RDS Postgres instances running Engine Version 10.15 (under Configuration Tab) and no Pending Maintenance is present (under Maintenance & Backup tab) for each one.

Complete an upgrade to this version of the code by:
checkout out this branch
Update update products/installation.yaml cro reference to point to -> quay.io/laurafitzgerald/cro-index:v0.31.0
Stop the operator and re-run using `INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make code/run`

During the upgrade Log into the AWS account and verify that
- each Postgres in the deployed is moved to EngineVersion 10.16
- each Postgres has applied any pending Service Updates present.
You should see each Postgres going into Upgrading and Maintenence Status

Verify that the config map `redhat-rhoam-installation-config` entry `cloud-resources` has a new entry for `STRATEGIES_CONFIG_MAP_NAME` with value `"cloud-resources-aws-strategies"`
Verify downtime by viewing the workload web app dashboard.

Verify that in the config map `"cloud-resources-aws-strategies` that there is now a serviceUpdate entry for Postgres. Before the upgrade there was no entry present.

### CODE REVIEW

**Comment by MStokluska** [comment]
image

**Comment by austincunningham** [comment]
Verified 
- installed 10.14 via olm
- confirmed the engine version is 10.15 
- built an index of this branch for 10.15 and updated the catalogSource
- upgrade completed
- engine version moved to 10.16 on the three rds instances
image
- once the upgrade was completed maintenance happend on aws console
- A system update was applied
- All rds instances became available 
image

/lgtm

File: controllers/rhmi/types.go [hunk 1]
```diff
@@ -25,13 +25,6 @@ var (
 					},
 				},
 			},
-			{
```
Comment (MStokluska) [inline controllers/rhmi/types.go:28]: @laurafitzgerald just so we won't forget about it - this needs to go back in and be removed in the next release.

**Comment by makslion** [comment]
I've watched an upgrade scenario happening. 
Starting with 10.15 RDS instances were updated to 10.16 and the upgrade process was going as expected except for the issue with setting `toVersion` to nill on the CRO complete (when the upgrade, technically, just started). This will be addressed later

**Comment by MStokluska** [comment]
e2e error log:
```
Error with version of cloud-resources operator deployed on cluster. Expected 0.32.0. Got 0.30.0
```

**Comment by MStokluska** [comment]
@austincunningham screenshot from A07 running locally:
image

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	github.com/3scale-ops/marin3r v0.8.0
-	github.com/3scale/3scale-operator v0.2.1-0.20210312114906-e612e89addbf
+	github.com/3scale/3scale-operator v0.7.0
```
Comment (MStokluska) [inline go.mod:7]: @laurafitzgerald do we know where is this coming from?

**Comment by MStokluska** [comment]
/lgtm 
Reviewed code, all looks good, functionality was reviewed together with Laura and Austin, all looks good. 
I suggest bypassing A07 failure as it  passes locally and failure in prow might be related to the amount of commits coming in this PR.

**Comment by MStokluska** [comment]
overridden e2e as we (@laurafitzgerald @briangallagher @austincunningham) have agreed to bypass the failed test A07 due to it passing locally.

**Pull request merged**